web/src/main/webapp/my-app/marketplace/controllers.js

```diff
@@ -58,9 +58,17 @@ define(['angular', 'jquery', 'require'], function(angular, $, require) {
           && portlet.portletName.indexOf('cms') != -1; // static content portlet
       };
 
+      $scope.isGuest = function(mainService.isGuest())
+          .then(function(isGuest) {
+            return isGuest;
+        }).catch(function() {
+          $log.warn('Cannot get isGuest');
+          return true;
+        });
+
       $scope.getLaunchURL = function(marketplaceEntry) {
         var layoutObj = marketplaceEntry.layoutObject;
-        if ($rootScope.GuestMode && !marketplaceEntry.hasInLayout) {
+        if ($scope.isGuest && !marketplaceEntry.hasInLayout) {
```

Comment:
:question: Is there ever a case where `$scope.isGuest` will falsy? It is assigned a function on 61 which will be truthy.